R/get.R

```diff
@@ -0,0 +1,228 @@
+#' Get treaty articles by number or match
+#'
+#' The function splits treaty texts into lists that reflect a structure
+#' based on having a preamble and several articles.
+#' Once articles are split, users can access the "preambles",
+#' "accession" and "termination" clauses, or "annexes"
+#' within a collection of treaty texts.
+#' The selection of each of these articles relies on regex matching
+#' certain expressions unique to each of them.
+#' Alternatively, users can get all the articles that contain a certain
+#' 'desired' word within a collection of treaty texts.
+#' As well, users can select only certain types of treaties for which
+#' specific articles are to be returned (e.g. agreements).
+#' @param textvar A text variable
+#' @param article Would you like to get a specific article?
+#' Null by default.
+#' Other options include the "preamble",
+#' "termination" clause, "membership" clause, or "annex".
+#' The specified portion for all treaties will be returned.
+#' @param match A regex match for a word(s) or expression.
+#' For multiple words, please use "|" to divide them.
+#' @param treaty_type What types of treaty do you want to look at?
+#' By default, "all".
+#' Other treaty types include: "agreements", "protocols", "amendments",
+#' "notes", "memorandum", and "resolutions".
+#' @details Please make sure treaty texts have been standardised first
+#' using `standardise_texts()` for best results.
+#' @importFrom purrr map_chr map
+#' @importFrom stringr str_extract str_replace_all str_trim str_split
+#' @importFrom stringi stri_trans_general
+#' @return A list of treaty sections of the same length.
+#' @examples
+#' \dontrun{
+#' t <- standardise_texts(sample(manyenviron::texts$AGR_TXT$Text, 30))
+#' get_articles(t)
+#' get_articles(t, article = "preamble")
+#' get_articles(t, article = "accession")
+#' get_articles(t, article = "termination")
+#' get_articles(t, article = "annex")
+#' get_articles(t, match = "constitution")
+#' get_articles(t, article = "preamble", match = "amend")
+#' get_articles(t, treaty_type = "agreements")
+#' get_articles(t, treaty_type = "protocols")
+#' get_articles(t, treaty_type = "amendments")
+#' }
+#' @export
+get_articles <- function(textvar, article = NULL,
+                         match = NULL, treaty_type = "all") {
+  usethis::ui_info("Please make sure treaty texts have been standardised first
+                   using `standardise_texts()`")
```

Comment:
Base this on a check to see whether the text has already been pre-configured/structured?